src/constants.ts

```diff
@@ -24,46 +25,65 @@ export const MAX_STORAGE_ITEM_SIZE = 256n;
 export const ADDR_BOUND = 2n ** 251n - MAX_STORAGE_ITEM_SIZE;
 
 const range = (min: bigint, max: bigint) => ({ min, max }) as const;
+
 export const RANGE_FELT = range(ZERO, PRIME - 1n);
 export const RANGE_I128 = range(-(2n ** 127n), 2n ** 127n - 1n);
 export const RANGE_U128 = range(ZERO, 2n ** 128n - 1n);
 
-export enum BaseUrl {
-  SN_MAIN = 'https://alpha-mainnet.starknet.io',
-  SN_GOERLI = 'https://alpha4.starknet.io',
-  SN_SEPOLIA = 'https://alpha-sepolia.starknet.io',
-}
-
-export enum NetworkName {
-  SN_MAIN = 'SN_MAIN',
-  SN_GOERLI = 'SN_GOERLI',
-  SN_SEPOLIA = 'SN_SEPOLIA',
-}
-
-export enum StarknetChainId {
-  SN_MAIN = '0x534e5f4d41494e', // encodeShortString('SN_MAIN'),
-  SN_GOERLI = '0x534e5f474f45524c49', // encodeShortString('SN_GOERLI')
-  SN_SEPOLIA = '0x534e5f5345504f4c4941', // encodeShortString('SN_SEPOLIA')
-}
-
-export enum TransactionHashPrefix {
-  DECLARE = '0x6465636c617265', // encodeShortString('declare'),
-  DEPLOY = '0x6465706c6f79', // encodeShortString('deploy'),
-  DEPLOY_ACCOUNT = '0x6465706c6f795f6163636f756e74', // encodeShortString('deploy_account'),
-  INVOKE = '0x696e766f6b65', // encodeShortString('invoke'),
-  L1_HANDLER = '0x6c315f68616e646c6572', // encodeShortString('l1_handler'),
-}
-
-export const enum feeMarginPercentage {
-  L1_BOUND_MAX_AMOUNT = 50,
-  L1_BOUND_MAX_PRICE_PER_UNIT = 50,
-  MAX_FEE = 50,
-}
+export const BaseUrl = {
+  SN_MAIN: 'https://alpha-mainnet.starknet.io',
+  SN_GOERLI: 'https://alpha4.starknet.io',
+  SN_SEPOLIA: 'https://alpha-sepolia.starknet.io',
+} as const;
+
+// eslint-disable-next-line @typescript-eslint/no-redeclare -- intentionally naming the variable the same as the type
```

Comment:
@ivpavici 
The eslint disable is here needed because of the limitations of `eslint no-redeclare rule`. They even recommended adding comments like this, but it can be painful since we have this occurrence in a lot of places now.
We have two options:
- Disable the `@typescript-eslint/no-redeclare` rule (which is not the deal breaker)
- Write comments where it's needed (technically the only valid case is when we use same name for type and const on purpose)

Reply:
@ivpavici Moreover, it would be nice that we do `enum` replacement in the `starknet/types` repo.

Reply:
Here's the PR on `types/js` :
https://github.com/starknet-io/types-js/pull/4

Reply:
@tabaktoni please advise :)